docs/minio-as-eucalyptus-object-storage.md

```diff
@@ -0,0 +1,120 @@
+```
+Endpoint:  http://172.31.21.31:9000  http://127.0.0.1:9000
+AccessKey: GFQX5XMP1DSTIQ8JKRXN
+SecretKey: aFLQegoeIgzF/hK+Xymba7JwSANfn98ANNcKXz+S
+Region:    us-east-1
+SqsARNs:
+
+Browser Access:
+   http://172.31.21.31:9000  http://127.0.0.1:9000
+
+Command-line Access: https://docs.minio.io/docs/minio-client-quickstart-guide
+   $ mc config host add myminio http://172.31.21.31:9000 GFQX5XMP1DSTIQ8JKRXN aFLQegoeIgzF/hK+Xymba7JwSANfn98ANNcKXz+S
+
+Object API (Amazon S3 compatible):
+   Go:         https://docs.minio.io/docs/golang-client-quickstart-guide
+   Java:       https://docs.minio.io/docs/java-client-quickstart-guide
+   Python:     https://docs.minio.io/docs/python-client-quickstart-guide
+   JavaScript: https://docs.minio.io/docs/javascript-client-quickstart-guide
+```
+
+Minio also comes with a helpful web UI which can be accessible once we have Minio running.
+
+Next step would be to see how we can add a new provider client in Eucalyptus. It is definitely easier that it sounds :)
+
+If we try to reset the value for ObjectStorage it should show us the available supported objectstorage S3 provider clients or these values are also available in the cloud-output.log where user-facing services are running,
+
+```
+# euctl -r objectstorage.providerclient
+euctl: error (ModifyPropertyValueType): Cannot modify \
+objectstorage.providerclient.providerclient new value \
+is not a valid value.  Legal values are: walrus,ceph-rgw,riakcs
+```
+
+So, now we have to add another provider client for Minio. Technically, we can probably use riakcs, if you have deployed Eucalyptus with packages, but I am not gonna do that in this case, since I already have source build cloud.
+
+This post assumes that you know how to build Eucalyptus from source and doesn't go into much detail. The purpose of this post is to show how to add a third-party object storage like Minio with Eucalyptus. But again, feel free to use package installation and use riakcs as provider client, use the Minio's endpoint and user credentials as described below.
+
+To add minio as a provider client, first we need to create a file called MinioProviderClient.java,
+
+```
+/**
+* Location: eucalyptus/clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/providers/s3/MinioProviderClient.java
+*/
+
+package com.eucalyptus.objectstorage.providers.s3;
+
+import com.eucalyptus.objectstorage.providers.ObjectStorageProviders.ObjectStorageProviderClientProperty;
+
+@ObjectStorageProviderClientProperty("minio")
+public class MinioProviderClient extends S3ProviderClient {
+}
+```
+
+Build and install all of eucalyptus source or just build the jars.
+
+Stop eucalyptus-cloud service and copy eucalyptus-object-storage-4.4.0.jar to /usr/share/eucalyptus directory where user-facing services are running. Restart eucalyptus-cloud service.
+
+If everything goes well, when the services are enabled, we can now check if the supported object storage providers again.
```

Comment:
Sentence `we can now check if the supported object storage providers again` is not clear. Please reword.